tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/command/CommandFulfiller.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.teradata.tempto.internal.fulfillment.command;
+
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.context.State;
+import com.teradata.tempto.fulfillment.RequirementFulfiller;
+import com.teradata.tempto.fulfillment.TestStatus;
+import com.teradata.tempto.fulfillment.command.Command;
+import com.teradata.tempto.fulfillment.command.CommandRequirement;
+
+import java.io.IOException;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+
+public abstract class CommandFulfiller<T extends CommandRequirement>
+        implements RequirementFulfiller
+{
+    private final Class<T> requirementClass;
+    private final Configuration configuration;
+    private final String configurationCommandPrefix;
+
+    CommandFulfiller(Class<T> requirementClass, Configuration configuration, String configurationCommandPrefix)
+    {
+        this.requirementClass = requirementClass;
+        this.configuration = configuration;
+        this.configurationCommandPrefix = configurationCommandPrefix;
+    }
+
+    @Override
+    public final Set<State> fulfill(Set<Requirement> requirements)
+    {
+        requirements.stream()
+                .filter(requirement -> requirement.getClass().isAssignableFrom(requirementClass))
+                .map(requirement -> (T) requirement)
+                .distinct()
+                .forEach(this::fulfill);
+
+        configuration.getStringList(configurationCommandPrefix)
```

Comment:
Move configuration parsing out of fulfiller. Create `Requirement` objects externally in the same place other suite-level `Requirement` object are created. Just treat configuration like yet another source of requirements in addition to annotations/`getRequirements` methods in test classes.